### PR TITLE
Update ci workflow

### PR DESCRIPTION
**Pull request by pruhesi**
## Proposed changes

_Description of the changes implemented by this PR, feel free to be as verbose as needed_

- fix linting errors
- remove reliance on docker container for github actions ci workflow
- remove package json 
- remove node-sass

## Types of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] New feature
- [ ] Bugfix
- [x] Quality of Life

## Checklist

_Put an `x` in the boxes that apply._

- [x] Linter passes (`yarn lint`)
- [x] Unit Test pass (`yarn jest`)
- [ ] I have added at least two useful tests for new component (if applicable)
- [ ] I have added at least one Storybook story (if applicable)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          path: ${{env.working-directory}}
+
+  test:
+    needs: clone
+    runs-on: self-hosted
+    env:
+      working-directory: ${{github.sha}}
+
+    steps:
+      - uses: actions/setup-node@v3
```
Comment (countnoobula) [inline .github/workflows/ci.yaml:27]: Loving the step away from docker to rather using setup-actions!

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  clone:
+    runs-on: self-hosted
```
Comment (countnoobula) [inline .github/workflows/ci.yaml:11]: If you want, you could try run this on the GHA runner?
We have 3,000 Actions minutes/month available 😄

**Review by countnoobula — APPROVED** [review]
#mclovin-it

**Review by tmuzanenhamo — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  clone:
+    runs-on: ubuntu-latest
```
Comment (pruhesi) [inline .github/workflows/ci.yaml:11]: im not sure i understand how this works 
is this setup on this repo because i didnt see it in the list of runners

Comment (countnoobula) [reply .github/workflows/ci.yaml:11]: The migration to minutes was okay. This keyword is defined in https://github.com/actions/virtual-environments#available-environments and uses the minutes one. It uses ubuntu-20.04 and the setup commands will run on that instance.

You can view the minutes balance at https://github.com/organizations/bonlineza/settings/billing
It's an ubuntu 2 core machine. We have 3000 minutes, and have so far used 6.

You can also view the available tags and active jobs at https://github.com/organizations/bonlineza/settings/actions/hosted-runners

Comment (pruhesi) [reply .github/workflows/ci.yaml:11]: I see 
I was being dumb 
i understand

**Comment by pruhesi** [comment]
Cant approve since its my pr 
Approved

**Pull request merged**